src/mame/machine/dec_lk201.cpp

```diff
@@ -549,25 +553,34 @@ void lk201_device::device_timer(emu_timer &timer, device_timer_id id, int param,
 		break;
 
 	case 2:
-		m_speaker->set_output_gain(0, 0);
-		m_speaker->set_state(0);
+		if (m_beep_state == 2)
+		{
+			m_beeper->adjust(attotime::never);
+
+			m_speaker->set_state(0);
+			m_beep_state = 0;  // ready for new beep.
+
+		} else
+			m_beeper->adjust(attotime::from_msec(20));
+		
```

Comment:
Please make the use of line breaks consistent - if you're not putting opening brace on the same line as flow control statement, don't put the closing brace on the same line as a flow control statement, either.